0.25.3 Release Notes (#401)
astronomer/docs: enterprise/v0.25/10_resources/01_release-notes.md

We're committed to testing all Astronomer Enterprise versions for scale, reliability and security on Amazon EKS, Google GKE and Azure AKS. If you have any questions or an issue to report, don't hesitate to [reach out to us](https://support.astronomer.io).

## v0.25.3

Release Date: June 30th, 2021

> ### Improved OpenShift Support

Author: @vishwas-astro I added this section based on your comment below. This is something we'll want to document outside of release notes in the future, but in the interest of getting this released I'm including the required setup here.

cc: @pgvishnuram and @andriisoldatenko @schnie for technical review

> We've made several changes to make it easier for organizations already running OpenShift to implement the Astronomer platform. Most importantly, you can now use your any ingress controller for the Astronomer platform. To utilize this functionality, you need to complete the following setup:

Reviewer: @pgvishnuram and @andriisoldatenko - please review this section ASAP as this PR needs to get merged tomorrow.

@pgvishnuram - Do we want to keep this section Openshift agnostic and just say that the authorization has been moved to auth proxy side cars which allows us to no longer be opinionated with our nginx ingress controller and allows customers to bring their own ingress controller.

Reviewer: Also @jwitz "you can now use your any ingress controller" does not sound correct. Can you fix the wording there please?

Reviewer 2: agree with @vishwas-astro, you can use OpenShift as an example.

We're committed to testing all Astronomer Enterprise versions for scale, reliability and security on Amazon EKS, Google GKE and Azure AKS. If you have any questions or an issue to report, don't hesitate to [reach out to us](https://support.astronomer.io).

## v0.25.3

Release Date: June 30th, 2021

### Improved OpenShift Support

> We've made several changes to make it easier for organizations already running OpenShift to implement the Astronomer platform. Most importantly, you can now use your any ingress controller for the Astronomer platform. To utilize this functionality, you need to complete the following setup:

Reviewer 3: ```suggestion
We've made several changes to make it easier for organizations already running OpenShift to install the Astronomer platform. Most importantly, you can now use your any ingress controller for the Astronomer platform. To utilize this functionality, you need to complete the following setup:
```

It'd be nice to add half a sentence of more context here. Why would someone want to use another ingress controller? @vishwas-astro What's the ultimate value-add? I'm envisioning a sentence that's like, `If your team has restrictions on NGINX, for example, you can now use your any ingress controller for the Astronomer platform` (uninformed suggestion, but something like that).

Author: I think it's less "restrictions on NGINX" and more "some platforms favor certain controllers by default, and you shouldn't have to implement a new one just to use Astronomer". Tried to make that more clear, and steered the focus away from OpenShift.

      proxy_connect_timeout                   15s;
      proxy_send_timeout                      600s;
      proxy_read_timeout                      600s;
      proxy_buffering                         off;
      proxy_buffer_size                       4k;
      proxy_buffers                           4 4k;
      proxy_max_temp_file_size                1024m;
      proxy_request_buffering                 on;
      proxy_http_version                      1.1;
      proxy_cookie_domain                     off;
      proxy_cookie_path                       off;
      proxy_redirect                          off;

  ## If we need to add any extra annotations below option can be used to
  ## pass those values
  extraAnnotations: {}
  #kubernetes.io/ingress.class: "nginx"
```

### Support for Kubernetes 1.19 and 1.20

The Astronomer platform is now compatible with Kubernetes 1.19 and 1.20. As an Enterprise user, you can now upgrade your clusters to either of these two versions and take advantage of the latest Kubernetes features. For more information, refer to [Kubernetes 1.20 release notes](https://kubernetes.io/blog/2020/12/08/kubernetes-1-20-release-announcement/).

> ### Bypass User Invite Emails via Houston API

Reviewer 3: ```suggestion
### Bypass Email Verification for User Invites via Houston API
```

The Houston API `workspaceAddUser` mutation now includes a `bypassInvite` field. When this field is set to true, users invited to a Workspace no longer need to first verify their email addresses before accessing the Workspace. This type of query can be useful to minimize friction when programmatically inviting many users to your platform. For more information, see [Sample Mutations](/docs/enterprise/v0.25/manage-astronomer/houston-api#sample-mutations).

### Minor Improvements

> - Added support for installing Astronomer via ArgoCD, which facilitates continuous delivery for Kubernetes applications. For more information, read the [Astronomer Forum].

Reviewer 3: Presume we'll add link tomo?

> - Workspace Admins can now perform CRUD operations on any Deployment within their Workspace, even if they don't have Deployment Admin permissions for the given Deployment.

Reviewer 3: ```suggestion
- Workspace Admins can now see and perform CRUD operations on any Deployment within their Workspace, even if they don't have Deployment Admin permissions for the given Deployment.
```

Reviewer: dont think this is needed. CRUD means being able to Read/See.

- Prevented `NginxIngressHigh4XXRate` and `NginxIngressHigh5XXRate` alerts from over-firing during periods of low traffic.
- Added the ability to use non-RFC address spaces for Alertmanager.
- Added support for using Workload Identity to configure a GCP registry backend.
- Changed sidecar naming convention from `nginx` to `auth-proxy`.
- Added `fsGroup` to the Webserver `securityContext` to enable [role assumption](https://docs.aws.amazon.com/eks/latest/userguide/security_iam_service-with-iam.html) for EKS 1.17.

> ### Bugfixes

Reviewer 3: Is the line-item we have in Cloud release notes still applicable?

`- Fixed an issue where long task execution logs did not appear in the Airflow UI.
`

Author: Ooh good catch, it's actually not applicable for Cloud either. This change was reverted according to a previous comment on this PR.

Reviewer: This change was reverted. But if it was there, it would have been applicable for both Cloud and Enterprise.

- Fixed an issue where private root CAs did not work due to an unmounted certificate.
- Fixed broken links to Deployments in alert emails.
- Fixed an issue where historical logs did not appear in the Astronomer UI.
- Fixed an issue where System Admins were unable to create Deployments.

> - Fixed a visual bug where some Deployments with only 1 Scheduler were shown has having 2 in the Astronomer UI.

Reviewer 3: ```suggestion
- Fixed a visual bug where some Deployments with only 1 Scheduler were shown as having 2 in the Astronomer UI.
```

> - Fixed a visual bug where users without Workspace Admin permissions had a non-functional Invite Users button in the Astronomer UI.

Reviewer 3: ```suggestion
- Fixed a visual bug where users without Workspace Admin permissions had a non-functional **Invite Users** button in the Astronomer UI.
```

We're committed to testing all Astronomer Enterprise versions for scale, reliability and security on Amazon EKS, Google GKE and Azure AKS. If you have any questions or an issue to report, don't hesitate to [reach out to us](https://support.astronomer.io).

## v0.25.3

Release Date: June 30th, 2021

> ### Support for Any Ingress Controller

Author: @vishwas-astro @andriisoldatenko I rewrote this to focus on the Ingress controller support vs. the OpenShift support.

Reviewer: @jwitz should we keep this section concise here and provide instructions on ingress/ auth side car set up as a separate section in our documentation and just link it from here?

Author: @vishwas-astro We definitely should. But I'm not quite sure where we'd want this information, and in the interest of getting release notes published today I think it's OK to publish instructions here first and add to another document later. I also think some of the new docs/ docs site features we're working on will be better suited for storing this info.

Created a GitHub issue to keep track of this: https://github.com/astronomer/docs/issues/405

You can now use any Ingress controller for your Astronomer platform. This is particularly useful if you're installing Astronomer onto platforms like OpenShift, which favor a specific type of Ingress controller, or if you want to take advantage of any features not available through the default NGINX Ingress controller.

> To use your own Ingress controller, you need to configure a proxy sidecar to run authentication tasks that are usually managed by the default NGINX Ingress controller. For example, to configure an Ingress controller for OpenShift:

Reviewer: Our default nginx ingress controller currently has two responsibilities: act as a reverse proxy as well as delegate to our authorization server to authorize incoming requests. To use your own Ingress controller, you will need to disable Astronomer's default nginx ingress controller, set it up to work with Astronomer's backend services and enable authSideCar configuration in order to continue to authorize incoming requests. For example, to use your own ingress controller for OpenShift, you would need to:

1. Annotate the Kubernetes secret containing your certificate so that your Ingress controller recognizes your Astronomer certificate:

    ```sh
    kubectl annotate secret/astronomer-tls kubed.appscode.com/sync="platform-release=astronomer"
    ```

2. Patch your Ingress operator to enable OpenShift routes to bind the same hostname to multiple routes on different namespaces:

    ```sh
    oc -n openshift-ingress-operator patch ingresscontroller/default --patch '{"spec":{"routeAdmission":{"namespaceOwnership":"InterNamespaceAllowed"}}}' --type=merge
    ```

> 3. Add an annotation to your network policy to connect Astronomer's backend services to your Ingress controller:

Reviewer: Rephrase as:
Add an annotation to your network policy to connect to Astronomer's backend services from your Ingress controller:

    ```sh
    kubectl label namespace/ingress-nginx network.openshift.io/policy-group=ingress
    ```

> 4. Configure the Astronomer Helm chart to deploy an auth sidecar for running your own Ingress controller. That configuration might look something like this:

Reviewer: @pgvishnuram or @pritt20 can you help fix the wording here? We are not deploying an auth side car on the Ingress controller. We are deploying the auth side car on any services that need authorization to be performed such as Kibana, Grafana etc and even the Airflow webservers.

Author: @vishwas-astro I think my current wording might resolve the miscommunication. My understanding now is that we need a sidecar to run some tasks that would have been managed by the NGINX controller, and need to be distributed when an org chooses a different controller.

> 4. Configure the Astronomer Helm chart to deploy a sidecar for authenticating to various components of the Astronomer platform. That configuration might look something like this:

Reviewer: The 3 main pieces that users would need to do to disable our default nginx ingress shipped with the platform and enable authSidecar for authorization are:
nginxEnabled: false
sccEnabled: true
authSidecar:
   enabled: true
   ... and the rest of that config section

@jwitz do you think we should highlight those parts within this yaml

    ```yaml
    global:
      baseDomain: <base-domain>

>       helmRepo: "https://pgvishnuram.github.io/airflow_sidecar_release"

Reviewer: Can we leave out the reference to @pgvishnuram 's private repo and change it to something generic?

      baseDomain: <base-domain>
      helmRepo: "https://github.io/company-helm-repo"
      tlsSecret: astronomer-tls
      nginxEnabled: false # Disable nginx ingress
      nodeExporterEnabled: false
      sccEnabled: true # Required for OpenShift
      postgresqlEnabled: true

>       authSidecar:  # Deploy auth sidecar to use different Ingress controller

Reviewer: Auth side car is not using an ingress controller. Auth side car is a sidecar container that authorizes the request and forwards it on to the service it is attached to.

Author: Copied that language from [here](https://www.notion.so/astronomerio/Open-shift-NGINX-sidecar-648ce0eb9d7d494ab7c2e839402c860c#6a651f96989a46b885a9cf814f510c9c), I'm going to hotfix this entire comment out as the configuration of the sidecar itself is plenty emphasis. Can't think of a way to comment that line that isn't redundant to the values.

We're committed to testing all Astronomer Enterprise versions for scale, reliability and security on Amazon EKS, Google GKE and Azure AKS. If you have any questions or an issue to report, don't hesitate to [reach out to us](https://support.astronomer.io).

## v0.25.3

> Release Date: June 25, 2021

Reviewer: Date is Jun 30th

> ### Support for Kubernetes 1.19 and 1.2

Reviewer: 1.20, not 1.2

> The Astronomer platform is now compatible with Kubernetes 1.19 and 1.2. Enterprise users can now configure their clusters to take advantage of the latest Kubernetes features as described in the [official documentation](https://kubernetes.io/blog/2020/12/08/kubernetes-1-20-release-announcement/).

Reviewer 3: ```suggestion
The Astronomer platform is now compatible with Kubernetes 1.19 and 1.20. As an Enterprise user, you can now upgrade your clusters to either of these two versions and take advantage of the latest Kubernetes features. For more information, refer to [Kubernetes 1.20 release notes](https://kubernetes.io/blog/2020/12/08/kubernetes-1-20-release-announcement/).
```

To learn more about what versions of key platform components are supported for any given release, read [Version Compatibility Reference](/docs/enterprise/v0.25/resources/version-compatibility-reference).

### Bypass User Invite Emails via Houston API

The Houston API `workspaceAddUser` mutation now includes a `bypassInvite` field. When this field is set to true, users invited to a Workspace no longer need to first verify their email addresses before accessing the Workspace. This type of query can be useful to minimize friction when programmatically inviting many users to your platform. For more information, see [Sample Mutations](/docs/enterprise/v0.25/manage-astronomer/houston-api#sample-mutations).

### Minor Improvements

- Workspace Admins can now perform CRUD operations on any Deployment within their Workspace, even if they don't have Deployment Admin permissions for the given Deployment.
- Added the ability to use non-RFC address spaces for Alertmanager.
- Added support for using Workload Identity to configure a GCP registry backend.

> - Changed sidecar naming convention from `nginx` to `auth-proxy`.

Reviewer: There is one huge feature that is missing from the docs & release notes. It is because we never wrote it up clearly in a GH issue.

Problem described in Issue 1 here: https://www.notion.so/astronomerio/AIP-Openshift-Experience-Improvement-7aab2a8d382a43e7b9c99200d23692e4

And the solution described here: https://www.notion.so/astronomerio/Open-shift-NGINX-sidecar-648ce0eb9d7d494ab7c2e839402c860c#6a651f96989a46b885a9cf814f510c9c

cc: @pgvishnuram and @andriisoldatenko who can help you document this along with @schnie

- Added `fsGroup` to the Webserver `securityContext` to enable [role assumption](https://docs.aws.amazon.com/eks/latest/userguide/security_iam_service-with-iam.html) for EKS 1.17.

> - Changed the default Scheduler count for new Deployments to 2.

Reviewer: Please see my above comment on this improvement. Same applies here as well.

> - Improved the experience of installing the Astronomer Helm Chart via ArgoCD. For more information, read the [Astronomer Forum].

Reviewer: Forum link?

Author: Has yet to be published, draft is here: 

> Because ArgoCD doesn't support sync wave dependencies for [app of apps](https://argoproj.github.io/argo-cd/operator-manual/cluster-bootstrapping/#app-of-apps-pattern) structures, you'll need to do a slight workaround when installing Astronomer using ArgoCD:

> 1. Create a new app as you normally would, making sure to specify `enableArgoCDAnnotation: true` in your Astronomer Helm Chart. 
> 2. Run a sync with every component of the Astronomer platform selected. Once you see `astronomer-houston-db-migrations`, terminate the sync.
> 3. Run a second sync, this time deselecting `astronomer-alertmanager` while keeping all other components selected. Wait for this sync to finish completely.
> 4. Run another sync with every component of the Astronomer platform selected. This sync should work, and your platform should be installed as expected.

Reviewer 3: I feel like this is less that we "improved the experience" and moreso that we're enabling an automation option for those interested - would tweak wording here accordingly. Might also add a quick 5-word description of Argo.

### Bugfixes

> - Fixed an issue where private root CA's did not work due to an unmounted certificate.

Reviewer 3: ```suggestion
- Fixed an issue where private root CAs did not work due to an unmounted certificate.
```

- Fixed broken links to Deployments in alert emails.
- Fixed an issue where historical logs did not appear in the Astronomer UI.

> - Fixed an issue where long task execution logs did not appear in the Airflow UI.

Reviewer: I think we can live this out as we had to revert that fix. @bote795 and @danielhoherd - please let @jwitz know what he can add in the release notes, if any.

Reviewer 4: Is this related to the concat plugin? https://github.com/astronomer/issues/issues/3113 ?

Reviewer: yes, this was the concat plugin fix from @bote795 which we had to revert.
